Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,91 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        private readonly Stream source;
+        private StreamWriter? streamWriter;
+        private StreamReader? streamReader;
+
         public DataLoader(Stream source)
         {
+            this.source = source;
         }
 
-        public List<Mailbox> Load()
+        public List<Mailbox>? Load()
         {
+            source.Position = 0;
+            streamReader = new StreamReader(source);
+            List<Mailbox> mailboxList;
+
+            try
+            {
+                mailboxList = JsonConvert.DeserializeObject<List<Mailbox>>(streamReader.ReadToEnd());
+            }
+            catch (JsonReaderException)
+            {
+                return null;
+            }
+            
+
+            if(mailboxList == null)
+            {
+                return new List<Mailbox>();
+            }
+
+            List<Mailbox> mailboxes = new Mailboxes(mailboxList, 30, 10);
+            return mailboxes;
         }
 
         public void Save(List<Mailbox> mailboxes)
         {
-            
+            source.Position = 0;
+            streamWriter = new StreamWriter(source);
```

Comment:
This could be declared as a local variable:
```suggestion
            using var streamWriter = new StreamWriter(source, leaveOpen:true);
```